### PR TITLE
 	Allow scenes to be reloaded dynamically through reset event

### PR DESCRIPTION
**Pull request by favreau**

### CODE REVIEW

**Comment by favreau** [comment]
Retest this please

File: plugins/engines/optix/OptiXScene.cpp [hunk 1]
```diff
@@ -48,34 +49,116 @@ OptiXScene::OptiXScene(
...
+    Scene::reset();
+
+    // Geometry
+    for( auto geometryInstance: _geometryInstances)
+        geometryInstance->destroy();
+    _geometryInstances.clear();
+
+    if( _geometryGroup )
+        _geometryGroup->destroy();
+    _geometryGroup = nullptr;
```
Comment (rdumusc) [inline plugins/engines/optix/OptiXScene.cpp]: This function could be made much simpler by wrapping all the destroyable objects in unique_ptrs with a custom deleter that calls destroy()

File: brayns/common/scene/Scene.cpp [hunk 2]
```diff
@@ -46,8 +46,14 @@ Scene::Scene(
 
 Scene::~Scene( )
 {
+    reset();
```
Comment (favreau) [inline brayns/common/scene/Scene.cpp]: Not really. It is necessary in the inherited classes, for the various engines, but definitely not required in the Abstract Scene class. Removed.

File: brayns/common/camera/Camera.cpp [hunk 3]
```diff
@@ -51,10 +51,13 @@ struct Camera::Impl
...
         const Vector3f& upVector )
     {
+        const Vector3f& p = getPosition();
+        const Vector3f& t = getTarget();
+        const Vector3f& u = getUpVector();
         setPosition( position );
         setTarget( target );
         setUpVector( upVector );
-        modified = true;
+        modified = ( p != position || t != target || u != upVector );
```
Comment (rdumusc) [inline brayns/common/camera/Camera.cpp]: yes, then the problem should be addressed in each of the individual setter, they must do (example code):
```
setPosition( newPos ){
if( newPos == _pos )
    return();
_pos = newPos;
}
```

Comment (favreau) [inline brayns/common/camera/Camera.cpp]: Sounds good. Done.

File: plugins/engines/optix/OptiXScene.cpp (hunk 1 above)
Comment (favreau) [inline plugins/engines/optix/OptiXScene.cpp]: Tried it but in the end, it was less readable.

**Pull request merged**